### PR TITLE
add some options

### PR DESCRIPTION
**Pull request by mizunashi-mana**
Add `force`, output modes and without-option options.

And, outputs make colorize.

### CODE REVIEW

**Comment by znz** [comment]
I have some questions.
- Why remove `set -u` ?
- When `git branch` returns a line includes `HEAD detached` ? I tried `git remote add mizunashi-mana https://github.com/mizunashi-mana/anyenv-update; git fetch mizunashi-mana; git checkout mizunashi-mana/add-some-options` and `git branch`, but it returns `* (detached from mizunashi-mana/add-some-options)` line. (tried git version is 1.9.1 on Ubuntu 14.04)
- Why add `--no-rebase --ff` to `git pull` ?

**Comment by mizunashi-mana** [comment]
## Why remove `set -u`?

I think u option is not necessary for production.  This option is useful for debug, but is not helpful in running.  In bash case, we use many env vars and they are fluctuating.  I feel flexible runnning is needed for bash script.

## When git branch returns a line includes HEAD detached ?

Oh, sorry... It seems my mistake and less knowledge.  However, I have no good idea.  Are there any suggestions?

If no suggestion is given, I use `git name-rev` to fix.

## Why add --no-rebase --ff to git pull ?

The action of `git pull` is different for each environments. (Ref: https://git-scm.com/docs/git-config pull section) `--no-rebase --ff` means using fast fowarded merge for pull action.  If there are non fast-forwarded commits, `git pull` starts editor for merge, force rebase or does anything by each environments.  `--no-rebase --ff` just runs fast forwarded merge if the env repo has clean status, and doesn't do anything when non fast forwarded commits. (For example, anyone do force push to remote env repo or user was changed own local env repo.)

**Comment by znz** [comment]
Thank you for your explanations.

I think `set -u` prevents some bugs. (e.g. https://github.com/dokku/dokku/commit/722564b2da81b64059971c28bfac1b404c779223 )
Then I like to use it. But ok to remove it. I will try to use another solution. (e.g. `shellcheck`)

I think that treatment of `HEAD detached` can be omitted. I think that who edit files under `~/.anyenv` should know what it means. But if another fix is exist, it is welcome.

Who changed default behavior of `git pull` may want that behavior, but `--no-rebase --ff` means that this plugin wants default behavior. I think that it is good behavior.

**Comment by mizunashi-mana** [comment]
Thanks for review!

I delete check logic for `detached HEAD`.

**Comment by mizunashi-mana** [comment]
Any news?

**Comment by znz** [comment]
I updated to use shellcheck.
Please rebase (and update something if required) and `push -f` to run shellcheck on travis ci.

**Comment by mizunashi-mana** [comment]
Thanks for greate update!

I rebased and fix some misses.

File: bin/anyenv-update [hunk 1]
```diff
@@ -1,38 +1,195 @@
...
+    echo git "$@" | indent_output
+  elif [ -n "$QUIET" ]; then
+    # mute all
+    command git "$@" &>/dev/null
+  elif [ -n "$VERBOSE" ]; then
+    # output all
+    command git "$@" 2>&1 | indent_output
+  else
+    # output only stderr
+    command git "$@" >/dev/null 2>&1 | indent_output
```
Comment (znz) [inline bin/anyenv-update]: `output only stderr` is intended behavior, use `shellcheck disable`

```
diff --git a/bin/anyenv-update b/bin/anyenv-update
index 3cab33a..86a9375 100755
--- a/bin/anyenv-update
+++ b/bin/anyenv-update
@@ -87,7 +87,8 @@ git () {
     command git "$@" 2>&1 | indent_output
   else
     # output only stderr
-    command git "$@" >/dev/null 2>&1 | indent_output
+    # shellcheck disable=SC2069
+    command git "$@" 2>&1 >/dev/null | indent_output
   fi
 }
```

**Comment by mizunashi-mana** [comment]
@znz
Thanks for your check and accurate advice.
I updated!

**Comment by znz** [comment]
Merged. Thanks!

**Pull request merged**